### DISCUSSION TITLE
Set alpha to opaque without changing RGB on an sRGB png

### DISCUSSION THREAD

**Discussion by dcsobral** [ImageMagick/ImageMagick#3142] [Help]
I have a sRGB 8-bit PNG file with these colors:
```
 convert splat4.png -format %c histogram:info:-
  29797353: (  0,  0,  0,  0) #00000000 none
       984: (  0,  3,  0,  0) #00030000 srgba(0,3,0,0)
       815: (  0,  5,  0,  0) #00050000 srgba(0,5,0,0)
       733: (  0,  6,  0,  0) #00060000 srgba(0,6,0,0)
       672: (  0,  8,  0,  0) #00080000 srgba(0,8,0,0)
       692: (  0, 10,  0,  0) #000A0000 srgba(0,10,0,0)
       667: (  0, 12,  0,  0) #000C0000 srgba(0,12,0,0)
       644: (  0, 14,  0,  0) #000E0000 srgba(0,14,0,0)
       692: (  0, 16,  0,  0) #00100000 srgba(0,16,0,0)
       745: (  0, 19,  0,  0) #00130000 srgba(0,19,0,0)
       693: (  0, 20,  0,  0) #00140000 srgba(0,20,0,0)
       781: (  0, 23,  0,  0) #00170000 srgba(0,23,0,0)
       852: (  0, 25,  0,  0) #00190000 srgba(0,25,0,0)
       907: (  0, 27,  0,  0) #001B0000 srgba(0,27,0,0)
   7941506: (  0, 28,  0,  0) #001C0000 srgba(0,28,0,0)
```
I need to set the alpha channel to opaque on all of them, and then perform some other operations on it, but everything I tried changes the RGB components of the pixels.  For example:
```
convert splat4.png -alpha off -format %c histogram:info:-
  29796709: (  0,  0,  0) #000000 black
       644: (  0,  1,  0) #000100 srgb(0,1,0)
       523: (  0,  2,  0) #000200 srgb(0,2,0)
(etc)
```
I cannot have that, I need every pixel to have the exact same RGB components, but with alpha set to 100%. How can I do it?

[Download link for splat4.png](https://www.dropbox.com/s/adc3x80gq4yeanl/splat4.png?dl=0)

**Comment by urban-warrior** [comment]
Can you post a link to splat4.png.  We need to download it and reproduce the problem before we can offer any useful advice.

**Reply by dcsobral** [reply]
Edited the question with a link to splat4.png.

**Comment by fmw42** [comment]
Try

`convert  splat4.png -transparent white -alpha off -format %c histogram:info:-`

**Reply by dcsobral** [reply]
Same result.

**Comment by urban-warrior** [comment]
Dealing with alpha is tricky.  For fully transparent pixels, the color component is ignored as fully transparent pixels have no color.  However, we added a patch so that instead of comparing pixel color, we compare individual pixel components.  With the patch, we get expected results.  Look for the patch in the next day or two.
